tests/e2e/test_run_demos.py

```diff
@@ -0,0 +1,195 @@
+import logging
+import os
+
+from datetime import timedelta
+from pathlib import Path
+from databricks.sdk import WorkspaceClient
+from databricks.sdk.service.workspace import ImportFormat
+from databricks.sdk.service.pipelines import NotebookLibrary, PipelinesEnvironment, PipelineLibrary
+from databricks.sdk.service.jobs import NotebookTask, PipelineTask, Run, Task, TerminationTypeType
+
+logging.getLogger("tests").setLevel("DEBUG")
+logging.getLogger("databricks.labs.dqx").setLevel("DEBUG")
+logger = logging.getLogger(__name__)
+
+RETRY_INTERVAL_SECONDS = 30
+TEST_LIBRARY_REF = f"git+https://github.com/databrickslabs/dqx.git@refs/pull/{os.getenv('REF_NAME')}"
```

Comment:
This will not work for nightly builds as they run on main branch. The url should be fetched dynamically. Example:
```
import subprocess

def get_git_url(repo_path="."):
    """Fetch the Git remote URL."""
    return subprocess.check_output(
        ["git", "-C", repo_path, "config", "--get", "remote.origin.url"],
        text=True
    ).strip()

def get_current_branch(repo_path="."):
    """Fetch the current branch name."""
    return subprocess.check_output(
        ["git", "-C", repo_path, "rev-parse", "--abbrev-ref", "HEAD"],
        text=True
    ).strip()

git_url = get_git_url()
current_branch = get_current_branch()

TEST_LIBRARY_REF = f"git+{git_url}@{current_branch}"

print(TEST_LIBRARY_REF)
```